metric/metric_test.go

```diff
@@ -42,7 +42,7 @@ func TestAddMetric(t *testing.T) {
 	}
 
 	for _, tc := range testCases {
-		t.Run(tc.name, func(t *testing.T) {
+		t.Run(tc.name, func(_ *testing.T) {
 			tMetricData := Data{
 				Name:  "RuntimeTest",
 				Type:  tc.metricType,
```

Comment:
If you instead move the initialization of the gomock controller and the clients into the subtest, you will get more accurate tests

Reply:
@nhhagen

Reply:
updated